pkg/store/consul.go

```diff
@@ -91,11 +105,55 @@ func (s *Consul) Get(key string) (*KVPair, error) {
 }
 
 // Put a value at "key"
-func (s *Consul) Put(key string, value []byte) error {
-	p := &api.KVPair{Key: s.normalize(key), Value: value}
-	if s.client == nil {
-		log.Error("Error initializing client")
+func (s *Consul) Put(key string, value []byte, opts *WriteOptions) error {
+
+	key = s.normalize(key)
+
+	p := &api.KVPair{
+		Key:   key,
+		Value: value,
+	}
+
+	if opts != nil && opts.Ephemeral {
```

Comment:
Since the session shares the same `ttl` as everyone else, I think you can create just one single session for all ephemeral keys.

Also, I would put this code in something like `createEphemeralSession` so you can remove it from the put.

Maybe you can call that function if `opts.Epheneral` and let it check if it needs or not to create a session:
```
if opts != nil && opts.Ephemeral {
   s.createEphemeralSession()
   [...]
```

createEphemeralSession could use a mutex to avoid creating multiple sessions for nothing.